commit d76d6d0ce93da902151550a8727740721ade1af3
Author: Wang YanQing <udknight@gmail.com>
Date:   Sun Feb 12 10:46:55 2017 +0800

    perf scripting perl: Fix compile error with some perl5 versions
    
    commit d7dd112ea5cacf91ae72c0714c3b911eb6016fea upstream.
    
    Fix below compile error:
    
      CC       util/scripting-engines/trace-event-perl.o
      In file included from /usr/lib/perl5/5.22.2/i686-linux/CORE/perl.h:5673:0,
                       from util/scripting-engines/trace-event-perl.c:31:
      /usr/lib/perl5/5.22.2/i686-linux/CORE/inline.h: In function 'S__is_utf8_char_slow':
      /usr/lib/perl5/5.22.2/i686-linux/CORE/inline.h:270:5: error: nested extern declaration of 'Perl___notused' [-Werror=nested-externs]
              dTHX;   /* The function called below requires thread context */
    			     ^
      cc1: all warnings being treated as errors
    
    After digging perl5 repository, I find out that we will meet this
    compile error with perl from v5.21.1 to v5.25.4
    
    Signed-off-by: Wang YanQing <udknight@gmail.com>
    Acked-by: Jiri Olsa <jolsa@kernel.org>
    Link: http://lkml.kernel.org/r/20170212024655.GA15997@udknight
    Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/tools/perf/util/scripting-engines/Build b/tools/perf/util/scripting-engines/Build
index 6516e22..82d28c6 100644
--- a/tools/perf/util/scripting-engines/Build
+++ b/tools/perf/util/scripting-engines/Build
@@ -1,6 +1,6 @@
 libperf-$(CONFIG_LIBPERL)   += trace-event-perl.o
 libperf-$(CONFIG_LIBPYTHON) += trace-event-python.o
 
-CFLAGS_trace-event-perl.o += $(PERL_EMBED_CCOPTS) -Wno-redundant-decls -Wno-strict-prototypes -Wno-unused-parameter -Wno-shadow -Wno-undef -Wno-switch-default
+CFLAGS_trace-event-perl.o += $(PERL_EMBED_CCOPTS) -Wno-redundant-decls -Wno-strict-prototypes -Wno-unused-parameter -Wno-shadow -Wno-nested-externs -Wno-undef -Wno-switch-default
 
 CFLAGS_trace-event-python.o += $(PYTHON_EMBED_CCOPTS) -Wno-redundant-decls -Wno-strict-prototypes -Wno-unused-parameter -Wno-shadow
